tests/integ/modin/frame/test_align.py

```diff
@@ -0,0 +1,214 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.utils import assert_frame_equal
+from tests.integ.utils.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
+def test_align_basic_axis0(join):
+    native_df = native_pd.DataFrame(
+        [[1, 2, 3, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"]
+    )
+    native_other_df = native_pd.DataFrame(
+        [[10, 20, 30, 40], [60, 70, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    native_left, native_right = native_df.align(
+        native_other_df,
+        join=join,
+        axis=0,
+        limit=None,
+        fill_axis=0,
+        broadcast_axis=None,
+    )
+    df = pd.DataFrame(native_df)
+    other_df = pd.DataFrame(native_other_df)
+    left, right = df.align(other_df, join=join, axis=0)
+    assert_frame_equal(left, native_left)
+    assert_frame_equal(right, native_right)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
+def test_align_basic_with_nulls_axis0(join):
+    native_df = native_pd.DataFrame(
+        [[1, 2, np.nan, 4], [6, 7, 8, 9]], columns=["D", "B", "E", "A"]
+    )
+    native_other_df = native_pd.DataFrame(
+        [[10, 20, 30, np.nan], [60, np.nan, 80, 90], [600, 700, 800, 900]],
+        columns=["A", "B", "C", "D"],
+    )
+    native_left, native_right = native_df.align(
+        native_other_df,
+        join=join,
+        axis=0,
+        limit=None,
+        fill_axis=0,
+        broadcast_axis=None,
+    )
+    df = pd.DataFrame(native_df)
+    other_df = pd.DataFrame(native_other_df)
+    left, right = df.align(other_df, join=join, axis=0)
+    assert_frame_equal(left, native_left)
+    assert_frame_equal(right, native_right)
+
+
+@sql_count_checker(query_count=2, join_count=2)
+@pytest.mark.parametrize("join", ["outer", "inner", "left", "right"])
+def test_align_basic_with_all_null_col_axis0(join):
```

Comment:
```suggestion
def test_align_basic_with_all_null_row_axis0(join):
```
Seems like none of the columns are all nan? the last row is though!